### PR TITLE
zendesk.tickets (major update), zendesk.chats (removed)

### PR DESCRIPTION
**Pull request by DavidDurman**
- Old Zendesk tickets connector was not working.
- This is a major upgrade with updates to non-working components together with getting on-par with Zapier when it comes to triggers/actions.

Q: It's been under heavy discussions whether to move this connector under the `zendesk/` directory or leave it as is in `zendesktickets/`.

This PR also updates the `zendesk.chats` connector and fixes https://github.com/clientIO/appmixer-components/issues/1873

### CODE REVIEW

**Review by jirihofman — APPROVED** [review]

File: src/appmixer/zendesktickets/package-lock.json
Comment (jirihofman) [inline src/appmixer/zendesktickets/package-lock.json:1]: We don't use `package-lock.json` files yet. Please remove it.

**Review by jirihofman — CHANGES_REQUESTED** [review]

**Review by jirihofman — APPROVED** [review]

**Review by vtalas — COMMENTED** [review]
@DavidDurman  quotas are missing, there should be at least some quotas set

**Comment by DavidDurman** [comment]
The rate limits of Zendesk are very complicated and also depend on the account tier. I'm not sure how this can be implemented using our current quotas mechanism. See https://developer.zendesk.com/api-reference/introduction/rate-limits/#zendesk-suite-plan-limits

**Comment by vtalas** [comment]
- [ ] add at least some basic quotas

**Review by vtalas — CHANGES_REQUESTED** [review]
apply quotas in compoent.json files

**Review by vtalas — APPROVED** [review]

**Comment by RamBachkheti** [comment]
@vtalas: The backoffice displays four different tickets labeled 'Zendesk.' Could you please let me know which one I should use? Additionally, could you remove any duplicates to avoid confusion?

Screenshot 2024-10-21 at 6 41 13 AM

**Comment by vtalas** [comment]
@RamBachkheti  it't this one: 
image

we're going to remove deprecated version from the marketplace

**Comment by RamBachkheti** [comment]
Issue 1: [CreateUser] Getting "error:DatabaseConflict", please see the following image:
- Screenshot 2024-10-22 at 11 27 33 AM
<hr>

Issue 2: [UpdateOrganization] "Domain Names" should have validation message when domain formate doesn't match, please see the following image:
Screenshot 2024-10-22 at 11 26 22 AM

**Comment by DavidDurman** [comment]
@vtalas why is Zendesk requiring me to enter clientID/secret on QA? This wasn't the case before and the standard OAuth was working fine:

image

**Comment by vtalas** [comment]
> @vtalas why is Zendesk requiring me to enter clientID/secret on QA? This wasn't the case before and the standard OAuth was working fine:
> 
> image

@DavidDurman  it's due to bug on in the authub (https://github.com/clientIO/appmixer-core/issues/2640), when the `globasKeys` is set to `true`, it won't ask for the clientId & secret.

**Comment by DavidDurman** [comment]
@RamBachkheti do you have repro steps to Issue nr. 1? It's working fine for me with these values:

image

**Comment by vtalas** [comment]
> Issue 2: I don't think we have a way to define a validation for "comma separated list of domains". What do you think @vtalas .

Surely we have, you can use the `pattern`  (regex). On the other hand, the velation message is not pretty in that case 
image

Also even it's great to catch as many errors as possible at the design phase I don't think it's crucial to introduce a regex in this case. @DavidDurman  you decide. I would rather consider including this note  
image

https://developer.zendesk.com/api-reference/ticketing/organizations/organizations/#update-organization

**Pull request merged**